File: L933_Number_of_Recent_Calls/Solution.java
class RecentCounter {
    Queue<Integer> queue = null;

    public RecentCounter() {
        queue = new LinkedList();
    }
    
    public int ping(int t) {
        queue.add(t);
        while(queue.size()> 0 && (t - queue.peek()) > 3000){
            queue.pop();
        }
        return queue.size(); 
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */


class RecentCounter1{
    private Queue <Integer> = null;
    RecentCounter(){
        queue = new LinkedList();
    }
    public int ping(int t){
        queue.add(t);
        while((t-queue.peek())>3000 && queue.size()>0 ){
            queue.pop();
        }
    }
    return queue.size();
}

File: L933_Number_of_Recent_Calls/solution.js

var RecentCounter = function() {
    this.arr = [];
    this.start = 0;

};

/** 
 * @param {number} t
 * @return {number}
 */
RecentCounter.prototype.ping = function(t) {
    this.arr.push(t);
    while(this.arr.length>0 && this.arr[this.start]< (t-3000) ){
        this.start++;  // 相当于 pop
    }
    return this.arr.length - this.start;

};

/** 
 * Your RecentCounter object will be instantiated and called as such:
 * var obj = new RecentCounter()
 * var param_1 = obj.ping(t)
 */